controllers/runnerdeployment_controller.go

```diff
@@ -94,7 +97,14 @@ func (r *RunnerDeploymentReconciler) Reconcile(req ctrl.Request) (ctrl.Result, e
 		oldSets = myRunnerReplicaSets[1:]
 	}
 
-	desiredRS, err := r.newRunnerReplicaSet(rd)
+	desiredReplicas, err := r.determineDesiredReplicas(rd)
+	if err != nil {
+		log.Error(err, "Could not determine desired replicas")
+
+		return ctrl.Result{}, err
+	}
+
+	desiredRS, err := r.newRunnerReplicaSet(rd, *desiredReplicas)
```

Comment:
Probably we'd better modify RunnerReplicaSet controller to prefer scaling down inactive runner first, so that scale down and happen more responsively.

Please also see how a similar project does it: https://github.com/philips-labs/terraform-aws-github-runner/blob/88bc451e5e36358666a220d9c737ad6639da03a1/modules/runners/lambdas/runners/src/scale-runners/scale-down.ts#L79-L101